### PR TITLE
fix(agent): updating support for otlp datastore

### PR DESCRIPTION
**Pull request by xoscar**
This PR...

## Changes

-

## Fixes

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: cli/pkg/starter/starter.go [hunk 1]
```diff
@@ -100,6 +100,10 @@ func (s *Starter) StartAgent(ctx context.Context, endpoint, name, agentApiKey, u
 		ServerURL: endpoint,
 		APIKey:    agentApiKey,
 		Name:      name,
+		OTLPServer: agentConfig.OtlpServer{
```
Comment (xoscar) [inline cli/pkg/starter/starter.go:103]: This needs to be configurable from the CLI

File: server/app/app.go [hunk 2]
```diff
@@ -230,8 +230,8 @@ func (app *App) Start(opts ...appOption) error {
 
 	tracedbFactory := tracedb.Factory(tracesRepo)
 
-	eventEmitter := executor.NewEventEmitter(testDB, subscriptionManager)
-	registerOtlpServer(app, tracesRepo, runRepo, eventEmitter, dataStoreRepo, tracer)
+	// eventEmitter := executor.NewEventEmitter(testDB, subscriptionManager)
```
Comment (xoscar) [inline server/app/app.go:233]: We might need a flag to enable/disable this (?)

File: agent/collector/ingester.go [hunk 3]
```diff
@@ -28,13 +24,6 @@ func newForwardIngester(ctx context.Context, batchTimeout time.Duration, remoteI
 		traceCache:     remoteIngesterConfig.traceCache,
 	}
 
-	err := ingester.connectToRemoteServer(ctx)
```
Comment (xoscar) [inline agent/collector/ingester.go:31]: Having a connect here was failing the initialization and we are not connecting to a server "yet"

Comment (mathnogueira) [reply agent/collector/ingester.go:31]: We might need to disable all the removed code based on a flag. 

The removed code is important in the future when we decide to forward traces from the agent to an OTLP-compatible endpoint. But for now, I think we can remove it.

File: server/app/app.go (hunk 2 above)
Comment (mathnogueira) [reply server/app/app.go:233]: Yes. If we cant have it on cloud, we should disable it via configuration

**Review by mathnogueira — APPROVED** [review]

**Pull request merged**